src/services/activityReports.test.js

```diff
@@ -1955,4 +1962,77 @@ describe('Activity report service', () => {
       });
     });
   });
+  describe('handleSoftDeleteReport', () => {
+    let user;
+    let report;
+    let recipient;
+    let grant;
+    let goal;
+
+    beforeAll(async () => {
+      user = await User.create({
+        ...mockUserFour,
+        hsesUserId: faker.datatype.string(10),
+        id: faker.datatype.number({ min: 90000 }),
+      });
+      recipient = await createRecipient({});
+      grant = await createGrant({ recipientId: recipient.id });
+
+      report = await ActivityReport.create({
+        ...submittedReport,
+        userId: user.id,
+        lastUpdatedById: user.id,
+        submissionStatus: REPORT_STATUSES.APPROVED,
+        calculatedStatus: REPORT_STATUSES.APPROVED,
+        activityRecipients: [{ activityRecipientId: grant.id }],
+      });
+      goal = await Goal.create({
+        name: 'Test Goal',
+        createdVia: 'activityReport',
+        grantId: grant.id,
+        status: GOAL_STATUS.NOT_STARTED,
+      });
+      await Objective.create({
+        title: 'Test Objective',
+        goalId: goal.id,
+        status: GOAL_STATUS.NOT_STARTED,
+      });
+      await ActivityReportGoal.create({
+        activityReportId: report.id,
+        goalId: goal.id,
+      });
+    });
+
+    afterAll(async () => {
+      await ActivityReportGoal.destroy({ where: { activityReportId: report.id }, force: false });
+      await Objective.destroy({ where: { goalId: goal.id }, force: true });
+      await Goal.destroy({ where: { id: goal.id }, force: true });
+      await ActivityReport.destroy({ where: { id: report.id }, force: true });
+      await Grant.destroy({ where: { id: grant.id }, force: true, individualHooks: true });
+      await Recipient.destroy({ where: { id: recipient.id }, force: true });
+      await User.destroy({ where: { id: user.id }, force: true });
+    });
+
+    it('soft deletes goals and objectives associated with the report', async () => {
```

Comment:
Does it make sense to add a test for a goal that is used by more than one report to test that filter?

Reply:
good idea, sure: [39f744c](https://github.com/HHS/Head-Start-TTADP/pull/2681/commits/39f744ca24618728444e617b6808f2f715bebe9d)